### DISCUSSION TITLE
Table Builder: Select filter using attribute?

### DISCUSSION THREAD

**Discussion by jrmypttrsn** [filamentphp/filament#2893] [Help]
I have an attribute on my Operation model for displaying the city (this data is coming from a third party):
```php
public function city(): Attribute
{
    if ($this->physical_address_city === '') {
        return Attribute::get(fn() => $this->mailing_address_city);
    }

    return Attribute::get(fn() => $this->physical_address_city);
}
```
My users are asking for a 'City' filter on the datatable I've built to display Operations. Is it possible to build a Select filter based on this attribute?

**Comment by danharrin** [comment]
yes, you could add both columns to the `query()`? but you cant use the Eloquent attribute in a database query, for obvious reasons

**Reply by jrmypttrsn** [reply]
> but you cant use the Eloquent attribute in a database query, for obvious reasons

Right. 

To get unique values from each column I just chain distinct() to the query?
